src/mon/AuthMonitor.cc

```diff
@@ -1478,8 +1477,8 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
       }
     }
 
-    if (!valid_caps(ceph_caps, &ss)) {
-      err = -EINVAL;
+    map<string,bufferlist> encoded_caps;
+    if ((err = _check_and_encode_caps(ceph_caps, encoded_caps, ss)) < 0) {
```

Comment:
No need for those extra parentheses anymore:

```cpp
int main() {
    int err = 0;
    // ...
    if (err = -1; err < 0) {
      std::cout << "err" << std::endl;
    }
}
```

See: https://godbolt.org/z/zdfx3Yvra.

Reply:
Fixed